### ISSUE TITLE
Add configuration option to define max count of skipped fragments before throwing an error

### ISSUE THREAD

**Issue by rwlodkowski** [mangui/flashls#457]
In the case when `hls_fragmentloadskipaftermaxretry` is `true` and the playlist is live or very long it's desirable to get an error after a specified amount of load errors on subsequent fragments in a row. This option will allow us to reload the playlist from another source or just log an error to the user.

Current behavior is like this:

```
// config
hls_fragmentloadmaxretry : 1,
hls_manifestloadmaxretry : 1,
hls_fragmentloadskipaftermaxretry: true

// logs
loading playlist -> OK
loading fragment 1 -> failed
loading fragment 1 -> failed
skip fragment 1
loading fragment 2 -> failed
loading fragment 2 -> failed
skip fragment 2
loading fragment 3 -> failed
loading fragment 3 -> failed
skip fragment 3
and so on to the infinity. 
It will never raise an error even when we reach the last fragment 
on the playlist (when it's VOD)
```

This is also related with #368.

This is illustration of behaviour I would like to get:

```
// config
hls_fragmentloadmaxretry : 1,
hls_manifestloadmaxretry : 1,
hls_fragmentloadskipaftermaxretry: true,
hls_maxskippedfragments: 3

// logs
loading playlist -> OK
loading fragment 1 -> failed
loading fragment 1 -> failed
skip fragment 1
loading fragment 2 -> failed
loading fragment 2 -> failed
skip fragment 2
loading fragment 3 -> failed
loading fragment 3 -> failed
skip fragment 3 -> skipped fragment limit reached
loading fragment 4 -> failed
loading fragment 4 -> failed
I/O ERROR raised
```

In case when the next fragment after skip is successfully loaded the skip counter is reset, so it's only applicable to the cases when the errors occur in a row.

**Comment by mangui** [comment]
yes that could be interesting, PR welcome

**Comment by rwlodkowski** [comment]
OK, will take a look at that. 

Could you tell me where exactly is the logic for that? Any advices/warnings in this case?

**Comment by mangui** [comment]
fragment skipping logic is here
https://github.com/mangui/flashls/blob/dev/src/org/mangui/hls/loader/FragmentLoader.as#L529-L542
you can raise the standard frag loading error, as done here:
https://github.com/mangui/flashls/blob/dev/src/org/mangui/hls/loader/FragmentLoader.as#L545
the error is instantiated here : 
https://github.com/mangui/flashls/blob/dev/src/org/mangui/hls/loader/FragmentLoader.as#L478

**Comment by rwlodkowski** [comment]
Thanks @mangui

**Comment by rwlodkowski** [comment]
Closed by #461
